init.sh

```diff
@@ -23,6 +23,34 @@ DIR_LIB_RT="$BASE/framework/lib/test_generation/runtime"
 DIR_LIB_GRADLE="$BASE/framework/lib/build_systems/gradle"
 mkdir -p "$DIR_LIB_GEN" && mkdir -p "$DIR_LIB_RT" && mkdir -p "$DIR_LIB_GRADLE"
 
+################################################################################
+#
+# Utility functions
+#
+
+# Get time of last data modification of a file
+get_time_of_last_data_modification() {
+    local USAGE="Usage: get_status <file>"
+    if [ "$#" != 1 ]; then
+        echo "$USAGE" >&2
```

Comment:
This might be error prone: if the caller doesn't check the return code, it may process nonsensical data. Should the script exit (`exit 1`) if the function is called without an argument?